mlflow/types/schema.py

```diff
@@ -1329,3 +1330,116 @@ def __eq__(self, other) -> bool:
 
     def __repr__(self) -> str:
         return repr(self.params)
+
+
+def convert_dataclass_to_schema(dataclass):
```

Comment:
Q: this is error-prone, do you think it will be simpler if we just implement `get_schema()` method for those dataclasses and return schema directly?

Reply:
Discussed offline, this can be reused if user wants to extend dataclasses, and they don't need to implement their own get_schema function in such cases.